salt/modules/status.py

```diff
@@ -366,8 +366,12 @@ def nproc():
 
         salt '*' status.nproc
     '''
-    data = __salt__['cmd.run']('nproc')
+    cmd = {
+        'Linux': 'nproc',
+        'FreeBSD': 'sysctl -n hw.ncpu',
+    }
     try:
+        data = __salt__['cmd.run'](cmd[__grains__['kernel']])
```

Comment:
Good question, we could raise a CommandExecutionError from salt.exceptions.
But you don't want to do that in the grains, in the grains we just want to return an empty dict because that routine is run on all systems